gtsfm/loader/loader_base.py

```diff
@@ -122,10 +141,28 @@ def get_valid_pairs(self) -> List[Tuple[int, int]]:
             list of valid index pairs.
         """
         indices = []
-
         for idx1 in range(self.__len__()):
             for idx2 in range(self.__len__()):
                 if self.validate_pair(idx1, idx2):
                     indices.append((idx1, idx2))
 
         return indices
+
+    def get_valid_triplets(self) -> List[Tuple[int, int, int]]:
+        """Get the valid triplets of images.
+
+        Returns:
+            list of indices of valid triplets.
+        """
+        indices = []
+        for idx1 in range(self.__len__()):
```

Comment:
any way to do this with itertools, instead of a triple for loop?

Reply:
@ayushbaid I think I can mark this as resolved, since you switched to itertools?